packages/ra-core/src/controller/input/useReferenceInputController.ts

```diff
@@ -116,6 +116,7 @@ export const useReferenceInputController = <RecordType extends RaRecord = any>(
         reference,
         options: {
             enabled: currentValue != null && currentValue !== '',
+            meta,
```

Comment:
Why not pass it the other query options as well?

Reply:
@slax57 those are not compatible typewise. `queryOptions` is currently typed as `UseQueryOptions<{
        data: RecordType[];
        total?: number;
        pageInfo?: {
            hasNextPage?: boolean;
            hasPreviousPage?: boolean;
        };
    }>` i.e. providing data with pagination info (which matches the `getList` result of the data provider just fine).

Contrary query options of `useReference` are typed as `UseQueryOptions<RecordType[], Error>;`, i.e. just an array of records, but no pagination info.

So if we'd want to provide the full set of react-query options to `useReference` then we might need to add another prop to `useReferenceInputController` that's specific to the item call.  But I don't feel that's feasible

Reply:
Alright, let's keep it that way for now then.